library/summary.ml

```diff
@@ -81,7 +96,7 @@ let unfreeze_summaries ?(partial=false) { summaries; ml_module } =
   (* The unfreezing of [ml_modules_summary] has to be anticipated since it
    * may modify the content of [summaries] by loading new ML modules *)
   begin match !sum_mod with
-  | None -> anomaly (str "Undeclared ML-MODULES summary.")
+  | None -> ()
```

Comment:
can you update the comment

Reply:
Sorry, what do you want the comment to say?

Reply:
The comment says why None is impossible, that is ML module loading is anticipated hence the handler of each summary entry is available (when you unfreeze you first handle loading of cmxs, so that all summary handlers are set).

Reply:
I may have misunderstood, but I thought the comment was saying why these summaries are handled separately, not why `None` is impossible.

Reply:
In particular, it is still true that the unfreezing of [ml_modules_summary] has to be anticipated.

Reply:
Well, then it is still an error if that summary entry is not declared.
I guess we could enforce it exists statically, but in the meanwhile...

So I don't get the new code, I thought None was a reasonable value thanks to this patch, and hence I asked to update the comment.
Did you get that anomaly in your branch at some point? Do you still get it?

Reply:
There are two things IMHO.

1. The separate handling of the ML module summary, which is what the comment is about. This patch does not touch that.
2. The possibility of unfreezing a set of frozen summaries which does not include the ML modules summary. Previously this was impossible, now it is possible if you freeze `Interp` summaries, since the ML modules summary is `Synterp` staged.

So yes, `None` is now a reasonable value, but AFAICT, it is unrelated to the comment. The comment is about 1.

Reply:
Hum, I see, `unfreeze` is the same for both stages. I'm I understanding right that `None` is only valid in `Interp` stage and that the comment (and the ref) have to be in `Synterp`?

Reply:
I mean, at least `None -> assert (ml_module = None)` is sensible, no?

Reply:
Sorry, I was focused on the link with the comment, but now I see your point. This is `freeze`, so indeed we'd expect this case to still be impossible.

Reply:
Sorry, I was focused on the link with the comment. Now I see your point. We'd expect this case to still be impossible, indeed. The ML modules summary should have been declared first.

Reply:
Fixed, I put back the anomaly.